### PR TITLE
[to #348] refine content in README

### PR DESCRIPTION
**Pull request by zz-jason**
Signed-off-by: Jian Zhang <zjsariel@gmail.com>

### What problem does this PR solve? 

cannot get full navigation in 'README.md'

### What is changed and how it works?

restructured the content and move part of them to separate files.

### Check List 

Tests 

 - No code

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -76,153 +44,16 @@ public class Main {
...
-### Metrics Parameter
-
-#### tikv.metrics.enable
-- whether to enable metrics exporting
-- default: false
-
-#### tikv.metrics.port
-- the metrics exporting http port
-- default: 3140
+Find more demo in [KVRawClientTest](https://github.com/birdstorm/KVRawClientTest/)
```
Comment (marsishandsome) [inline README.md:47]: better to move from personal repo to shared repo

Comment (zz-jason) [reply README.md:47]: Agree. How about addressing this issue in the near future?

Comment (birdstorm) [reply README.md:47]: +1

**Review by marsishandsome — APPROVED** [review]
LGTM

File: README.md [hunk 2]
```diff
@@ -7,46 +7,17 @@ It is supposed to:
 + Talk to TiKV for reading/writing data and the resulted data is encoded/decoded just like what we do in TiDB.
```
Comment (iosmanthus) [inline README.md:7]: We may add a section named Overview here.

File: README.md (hunk 1 above)
Comment (marsishandsome) [reply README.md:47]: create an issue?

Comment (zz-jason) [reply README.md:47]: I created a example section in the document: https://github.com/tikv/client-java/issues/348#issuecomment-992077936

File: README.md [hunk 3]
```diff
@@ -7,46 +7,17 @@ It is supposed to:
...
 + Talk to Coprocessor for calculation pushdown
 
-## How to build
+## Quick Start
```
Comment (iosmanthus) [inline README.md:10]: We may rename this section to `Example`?

File: README.md (hunk 2 above)
Comment (zz-jason) [reply README.md:7]: these statements are the product capability definition of the java client. I think we can optimize it in the future once we have conducted a better one.

File: README.md (hunk 1 above)
Comment (iosmanthus) [reply README.md:47]: The README of [tokio](https://github.com/tokio-rs/tokio) might a great example for library

Comment (iosmanthus) [reply README.md:47]: We could add a example in the repository just like `client-go`

File: README.md (hunk 3 above)
Comment (zz-jason) [reply README.md:10]: I think quick is better than examples, it indicates that this is a quick task to get familiar with java client.

**Pull request merged**